mne/io/tests/test_raw.py

```diff
@@ -153,3 +161,77 @@ def test_annotation_property_deprecation_warning():
     assert len(w) is 0
     with pytest.warns(DeprecationWarning, match='by assignment is deprecated'):
         raw.annotations = None
+
+
+@pytest.fixture
+def _dummy_raw():
+    info = create_info(ch_names=10, sfreq=10., ch_types=None, montage=None,
+                       verbose=None)
+    raw = RawArray(data=np.random.RandomState(0).randn(10, 10),
+                   info=info, first_samp=10)
+    raw.info['meas_date'] = 1
+    return raw
+
+
+@pytest.fixture
+def _dummy_annotation():
+    from mne import Annotations
+    return Annotations(onset=[.5],
+                       duration=[.2],
+                       description=['dummy'],
+                       orig_time=1.5)
+
+
+def test_relation_between_raw_meas_date_and_annotations_orig_time_A(
+        raw=_dummy_raw(), annot=_dummy_annotation()):
+    """Test meas_time is set and orig_time is set.
+
+    clips the annotations based on raw.data and resets the annotation based on
+    raw.info['meas_date]
+    """
+    raw.set_annotations(annot)
+
+    assert raw.annotations.orig_time == 1
+    assert raw.annotations.onset[0] == 1
+
+
+def test_relation_between_raw_meas_date_and_annotations_orig_time_B(
+        raw=_dummy_raw(), annot=_dummy_annotation()):
```

Comment:
I have never seen fixtures used this way. Is it recommended? I worry that by instantiating them as the default args to functions yourself (rather than letting `pytest` do some magic to instantiate them) I suspect that you allocate memory for them right away when the `test_*` file is loaded. It would probably be clearer to give readable names to the fixtures directly (`dummy_raw` is fine) and do:
```
test_...(dummy_raw, dummy_annot):
   ...
```
Also these test names are a bit long. More verbosity is not always better. We generally stick to shorter names like `test_meas_date_orig_time()`.

Also is it possible to combine A/B/C/D into a single test? It's using 50 lines for something that looks like it could use 15. If we start doing this for all of our tests they will balloon like crazy.

Reply:
I made a proposal already I can push it for you to see. And if you don't like it, you can push.